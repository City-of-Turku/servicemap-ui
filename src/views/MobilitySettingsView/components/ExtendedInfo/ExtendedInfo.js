import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@mui/material';

const ExtendedInfo = ({ classes, intl, translations }) => {
  const text = (message, props = {}) => (
    <Typography
      variant="body2"
      aria-label={intl.formatMessage({
        id: message,
      })}
      {...props}
    >
      {intl.formatMessage({
        id: message,
      })}
    </Typography>
  );

  return (
    <div className={classes.container}>
      {text(translations.message1)}
      <ul>
        {translations.zones.map(item => (
          <li key={item}>
            {text(item)}
          </li>
        ))}
      </ul>
      {text(translations.message2, { className: classes.margin })}
      {text(translations.message3)}
    </div>
  );
};

ExtendedInfo.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  translations: PropTypes.objectOf(PropTypes.any),
};

ExtendedInfo.defaultProps = {
  translations: {},
};

export default ExtendedInfo;
