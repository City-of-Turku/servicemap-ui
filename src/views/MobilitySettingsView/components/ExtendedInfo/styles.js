export default theme => ({
  container: {
    margin: theme.spacing(2),
    textAlign: 'left',
    paddingBottom: theme.spacing(2),
  },
  margin: {
    marginBottom: theme.spacing(1),
  },
});
