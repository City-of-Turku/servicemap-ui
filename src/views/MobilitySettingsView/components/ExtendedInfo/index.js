import ExtendedInfo from './ExtendedInfo';

export default ExtendedInfo;
