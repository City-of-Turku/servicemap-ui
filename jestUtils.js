import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });