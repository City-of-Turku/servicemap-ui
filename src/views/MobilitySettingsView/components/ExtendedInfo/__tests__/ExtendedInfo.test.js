/* eslint-disable max-len */
// Link.react.test.js
import React from 'react';
import ExtendedInfo from '../index';
import { getRenderWithProviders } from '../../../../../../jestUtils';
import finnishTranslations from '../../../../../i18n/fi';

const mockProps = {
  translations: {
    message1: 'mobilityPlatform.info.parkingChargeZones.paragraph.1',
    message2: 'mobilityPlatform.info.parkingChargeZones.paragraph.2',
    message3: 'mobilityPlatform.info.parkingChargeZones.paragraph.3',
    zones: [
      'mobilityPlatform.info.parkingChargeZones.zone.1',
      'mobilityPlatform.info.parkingChargeZones.zone.2',
      'mobilityPlatform.info.parkingChargeZones.zone.3',
    ],
  },
};

const renderWithProviders = getRenderWithProviders({});

describe('<ExtendedInfo />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<ExtendedInfo {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<ExtendedInfo {...mockProps} />);

    const p = container.querySelectorAll('p');
    const list = container.querySelectorAll('li');
    expect(p[0].textContent).toContain(finnishTranslations['mobilityPlatform.info.parkingChargeZones.paragraph.1']);
    expect(p[4].textContent).toContain(finnishTranslations['mobilityPlatform.info.parkingChargeZones.paragraph.2']);
    expect(p[5].textContent).toContain(finnishTranslations['mobilityPlatform.info.parkingChargeZones.paragraph.3']);
    expect(list[0].textContent).toContain(finnishTranslations['mobilityPlatform.info.parkingChargeZones.zone.1']);
    expect(list[1].textContent).toContain(finnishTranslations['mobilityPlatform.info.parkingChargeZones.zone.2']);
    expect(list[2].textContent).toContain(finnishTranslations['mobilityPlatform.info.parkingChargeZones.zone.3']);
  });
});
