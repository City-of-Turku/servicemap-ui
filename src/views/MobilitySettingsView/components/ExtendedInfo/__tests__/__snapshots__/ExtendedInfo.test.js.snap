// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtendedInfo /> should work 1`] = `
<div>
  <div
    class="injectIntl(ExtendedInfo)-container-1"
  >
    <p
      aria-label="testiteksti"
      class="MuiTypography-root MuiTypography-body2"
    >
      testiteksti
    </p>
    <ul>
      <li>
        <p
          aria-label="Ensimmäinen vyöhyke (ydinkeskusta-alue): 3,00 €/tunti"
          class="MuiTypography-root MuiTypography-body2"
        >
          Ensimmäinen vyöhyke (ydinkeskusta-alue): 3,00 €/tunti
        </p>
      </li>
      <li>
        <p
          aria-label="Toinen vyöhyke : 1,50 €/tunti"
          class="MuiTypography-root MuiTypography-body2"
        >
          Toinen vyöhyke : 1,50 €/tunti
        </p>
      </li>
      <li>
        <p
          aria-label="Kolmas vyöhyke: 0,60 €/tunti"
          class="MuiTypography-root MuiTypography-body2"
        >
          Kolmas vyöhyke: 0,60 €/tunti
        </p>
      </li>
    </ul>
    <p
      aria-label="testiteksti 2"
      class="MuiTypography-root injectIntl(ExtendedInfo)-margin-2 MuiTypography-body2"
    >
      testiteksti 2
    </p>
    <p
      aria-label="testiteksti 3"
      class="MuiTypography-root MuiTypography-body2"
    >
      testiteksti 3
    </p>
  </div>
</div>
`;
